build/marketplace/MarketPlace.sol

```diff
@@ -82,6 +86,91 @@ contract MarketPlace {
         return true;
     }
 
+    /// @notice sets the address for a pool
+    /// @param p enum value of the principal token
+    /// @param u address of the underlying asset
+    /// @param m maturity (timestamp) of the market
+    /// @param a address of the pool
+    /// @return bool true if successful
+    function setPool(
+        uint8 p,
+        address u,
+        uint256 m,
+        address a
+    ) external authorized(admin) returns (bool) {
+        require(pools[u][m][p] == address(0), 'pool exists');
+        pools[u][m][p] = a;
+        return true;
+    }
+
+    /// @notice sells the PT for the PT via the pool
+    /// @param p enum value of the principal token
+    /// @param u address of the underlying asset
+    /// @param m maturity (timestamp) of the market
+    /// @param a amount of PT to swap
+    /// @return uint128 amount of PT bought
+    function sellPT(
```

Comment:
so let's think about these names on a protocol big-picture. do we want `sellPT` or `sellPrincipalToken`

in our protocols generally (including swivel) we have gone with longer expressive method names

something to discuss

also gets us away from the `PT` vs `Pt` kind of argument...